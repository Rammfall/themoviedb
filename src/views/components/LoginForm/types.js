import PropTypes from 'prop-types'

export const propTypes = {
  isSubmitting: PropTypes.bool.isRequired,
  status: PropTypes.string
}

export const defaultProps = {
  status: undefined
}
