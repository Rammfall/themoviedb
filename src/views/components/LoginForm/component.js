import React from 'react'
import { FormattedMessage, useIntl } from 'react-intl'
import { Button, Typography, Form as AntForm } from 'antd'
import { LockOutlined, UserOutlined } from '@ant-design/icons'
import { Form, Field } from 'formik'

import InputField from 'Views/components/shared/InputField'
import GuestLayout from 'Views/layouts/GuestLayout'
import PasswordIcon from 'Views/components/shared/PasswordIcon'

import { defaultProps, propTypes } from './types'

const LoginFormComponent = ({ isSubmitting, status }) => {
  const { formatMessage } = useIntl()

  return (
    <GuestLayout>
      <Form>
        <Typography.Title>
          <FormattedMessage id="application.theMovieDB" />
        </Typography.Title>
        <Field
          name="username"
          component={InputField}
          placeholder={formatMessage({
            id: 'page.login.form.username'
          })}
          prefix={<UserOutlined style={{ color: 'rgba(0,0,0,.25)' }} />}
        />
        <Field
          name="password"
          component={InputField}
          placeholder={formatMessage({
            id: 'page.login.form.password'
          })}
          iconRender={PasswordIcon}
          type="password"
          prefix={<LockOutlined style={{ color: 'rgba(0,0,0,.25)' }} />}
        />
        <AntForm.Item>
          <Button
            type="primary"
            htmlType="submit"
            loading={isSubmitting}
          >
            {formatMessage({ id: 'page.login.form.login' })}
          </Button>
        </AntForm.Item>
        <p>{status}</p>
      </Form>
    </GuestLayout>
  )
}

LoginFormComponent.propTypes = propTypes

LoginFormComponent.defaultProps = defaultProps

export default LoginFormComponent
