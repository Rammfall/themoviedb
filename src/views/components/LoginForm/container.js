import React, { Component } from 'react'
import { withFormik } from 'formik'
import * as yup from 'yup'
import LoginFormComponent from './component'

export const handleSubmit = (
  values,
  { props, setErrors, setSubmitting, setStatus, resetForm, setValues }
) => {
  props.onSubmit(
    values,
    setErrors,
    setSubmitting,
    setStatus,
    resetForm,
    setValues
  )
}

// eslint-disable-next-line react/prefer-stateless-function
class LoginForm extends Component {
  render() {
    return <LoginFormComponent />
  }
}

export default withFormik({
  validationSchema: yup.object().shape({
    username: yup.string().required(),
    password: yup.string().min(4).required()
  }),
  mapPropsToValues: () => ({
    username: '',
    password: ''
  }),
  handleSubmit
})(LoginForm)
