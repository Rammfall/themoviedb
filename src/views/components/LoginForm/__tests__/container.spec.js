import React from 'react'
import { shallow } from 'enzyme'

import LoginForm, { handleSubmit } from '../container'

describe('LoginForm to match snapshot', () => {
  it('with default props', () => {
    const container = shallow(<LoginForm />)
    const props = {
      onSubmit: jest.fn()
    }

    expect(
      handleSubmit({ email: 'tset', password: 'test' }, { props })
    ).toMatchSnapshot()
    expect(container).toMatchSnapshot()
  })
})
