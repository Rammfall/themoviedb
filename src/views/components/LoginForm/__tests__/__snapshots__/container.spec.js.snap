// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm to match snapshot with default props 1`] = `undefined`;

exports[`LoginForm to match snapshot with default props 2`] = `
<Formik
  handleSubmit={[Function]}
  initialValues={
    Object {
      "password": "",
      "username": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={[Function]}
>
  <Component />
</Formik>
`;
